packages/core/src/libraries/role-scope.ts

```diff
@@ -0,0 +1,63 @@
+import { isManagementApi, RoleType } from '@logto/schemas';
+
+import RequestError from '#src/errors/RequestError/index.js';
+import type Queries from '#src/tenants/Queries.js';
+import assertThat from '#src/utils/assert-that.js';
+
+export const createRoleScopeLibrary = (queries: Queries) => {
+  const {
+    roles: { findRoleById },
+    scopes: { findScopeById },
+    resources: { findResourceById },
+    rolesScopes: { findRolesScopesByRoleId },
+  } = queries;
+
+  const validateRoleScopeAssignment = async (
+    scopeIds: string[],
+    roleId: string,
+    skipScopeExistenceCheck = false
```

Comment:
nit: a boolean argument is sometimes confusing, how about an options object:

```
{ skipScopeExistenceCheck?: boolean }
```

Reply:
Updated.